core/src/main/java/com/scalar/db/transaction/autocommit/AutoCommitTransactionManager.java

```diff
@@ -0,0 +1,115 @@
+import com.scalar.db.service.StorageFactory;
+import java.util.UUID;
+
+public class AutoCommitTransactionManager
+    extends ActiveTransactionManagedDistributedTransactionManager {
+
+  private final DistributedStorage storage;
+
+  public AutoCommitTransactionManager(DatabaseConfig databaseConfig) {
+    super(databaseConfig);
+    StorageFactory storageFactory = StorageFactory.create(databaseConfig.getProperties());
+    storage = storageFactory.getStorage();
+  }
+
+  @VisibleForTesting
+  AutoCommitTransactionManager(DatabaseConfig databaseConfig, DistributedStorage storage) {
+    super(databaseConfig);
+    this.storage = storage;
+  }
+
+  @Override
+  public DistributedTransaction begin() throws TransactionException {
+    String txId = UUID.randomUUID().toString();
+    return begin(txId);
+  }
+
+  @Override
+  public DistributedTransaction begin(String txId) throws TransactionException {
+    AutoCommitTransaction transaction = new AutoCommitTransaction(txId, storage);
+    getNamespace().ifPresent(transaction::withNamespace);
+    getTable().ifPresent(transaction::withTable);
+    return decorate(transaction);
+  }
+
+  /** @deprecated As of release 2.4.0. Will be removed in release 4.0.0. */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
+  @Override
+  public DistributedTransaction start(Isolation isolation) throws TransactionException {
+    return begin();
+  }
+
+  /** @deprecated As of release 2.4.0. Will be removed in release 4.0.0. */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
+  @Override
+  public DistributedTransaction start(String txId, Isolation isolation)
+      throws TransactionException {
+    return begin(txId);
+  }
+
+  /** @deprecated As of release 2.4.0. Will be removed in release 4.0.0. */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
+  @Override
+  public DistributedTransaction start(Isolation isolation, SerializableStrategy strategy)
+      throws TransactionException {
+    return begin();
+  }
+
+  /** @deprecated As of release 2.4.0. Will be removed in release 4.0.0. */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
+  @Override
+  public DistributedTransaction start(SerializableStrategy strategy) throws TransactionException {
+    return begin();
+  }
+
+  /** @deprecated As of release 2.4.0. Will be removed in release 4.0.0. */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
+  @Override
+  public DistributedTransaction start(String txId, SerializableStrategy strategy)
+      throws TransactionException {
+    return begin(txId);
+  }
+
+  /** @deprecated As of release 2.4.0. Will be removed in release 4.0.0. */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
+  @Override
+  public DistributedTransaction start(
+      String txId, Isolation isolation, SerializableStrategy strategy) throws TransactionException {
+    return begin(txId);
+  }
+
+  @Override
+  public TransactionState getState(String txId) {
```

Comment:
[question] This reminds me that maybe this method signature should be moved from DistributedTransactionManager to ConsensusCommitDistributedTransactionManager or something in the future? I thought the same thing about `DistributedTransactionAdmin.createCoordinatorTables()`.